src/goose.rs

```diff
@@ -913,6 +910,153 @@ impl GooseUser {
+    /// checking.
+    ///
+    /// For a safe alternative see [`get_session_data`].
+    ///
+    /// # Safety
+    ///
+    /// Calling this method on an GooseUser without session data or with a different type `T` is *[undefined behavior]*
+    ///
+    /// # Example
+    /// ```rust
+    /// use goose::prelude::*;
+    ///
+    /// #[derive(Debug)]
+    /// struct Foo(String);
+    ///
+    /// let mut task = task!(get_session_data_uncheck_function);
+    ///
+    /// /// A very simple task that makes a GET request.
+    /// async fn get_session_data_uncheck_function(user: &mut GooseUser) -> GooseTaskResult {
+    ///     let foo = user.get_session_data_uncheck::<Foo>();
+    ///     println!("Session data: {}", foo.0);
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_session_data_uncheck<T: GooseUserData>(&self) -> &T {
+        let session_data = self.session_data.as_deref().expect("Missing session data!");
+
+        session_data
+            .downcast_ref::<T>()
+            .expect("Invalid session data!")
+    }
+
+    /// Returns a mutable reference to GooseUser's session data.
+    ///
+    /// Return none if no session data has been set or that the session data is not of type `T`
+    ///
+    /// # Example
+    /// ```rust
+    /// use goose::prelude::*;
+    ///
+    /// #[derive(Debug)]
+    /// struct Foo(String);
+    ///
+    /// let mut task = task!(get_mut_session_data_function);
+    ///
+    /// /// A very simple task that makes a GET request.
+    /// async fn get_mut_session_data_function(user: &mut GooseUser) -> GooseTaskResult {
+    ///     let foo = user.get_mut_session_data::<Foo>().expect("Missing session data!");
+    ///     foo.0 = "Bar".to_owned();
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_mut_session_data<T: GooseUserData>(&mut self) -> Option<&mut T> {
+        match &mut self.session_data {
+            Some(data) => data.downcast_mut::<T>(),
+            None => None,
+        }
+    }
+
+    /// Returns a mutable reference to GooseUser's session data, without doing any type checking.
+    ///
+    /// For a safe alternative see [`get_mut_session_data`].
+    ///
+    /// # Safety
+    ///
+    /// Calling this method on an GooseUser without session data or with a different type `T` is *[undefined behavior]*
+    ///
+    /// # Example
+    /// ```rust
+    /// use goose::prelude::*;
+    ///
+    /// #[derive(Debug)]
+    /// struct Foo(String);
+    ///
+    /// let mut task = task!(get_mut_session_data_uncheck_function);
+    ///
+    /// /// A very simple task that makes a GET request.
+    /// async fn get_mut_session_data_uncheck_function(user: &mut GooseUser) -> GooseTaskResult {
+    ///     let foo = user.get_mut_session_data_uncheck::<Foo>();
+    ///     foo.0 = "Bar".to_owned();
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_mut_session_data_uncheck<T: GooseUserData>(&mut self) -> &mut T {
```

Comment:
It seems the standard naming for helpers that don't do any checking and return a mutable is `_unchecked_mut()` not `mut_ .. _uncheck()`:
https://doc.rust-lang.org/std/pin/struct.Pin.html#method.get_unchecked_mut

For consistency with Rust's std lib please rename this to `get_session_data_unchecked_mut`.